e2e/LongHaul/device/SystemHealthTelemetry.cs

```diff
@@ -1,53 +1,43 @@
-        private static readonly string s_processName = s_currentProcess.ProcessName;
-
-        private static readonly PerformanceCounter s_processCpuCounter = new(
-            "Process",
-            "% Processor Time",
-            s_processName,
-            true);
-        private static readonly PerformanceCounter s_processWorkingSet = new(
-            "Process",
-            "Working Set",
-            s_processName,
-            true);
-        private static readonly PerformanceCounter s_processWorkingSetPrivate = new(
-            "Process",
-            "Working Set - Private",
-            s_processName,
-            true);
-        private static readonly PerformanceCounter s_processPrivateBytes = new(
-            "Process",
-            "Private bytes",
-            s_processName,
-            true);
-        private static readonly PerformanceCounter s_processBytesInAllHeaps = null;
-        //new PerformanceCounter(
-        //    ".NET CLR Memory",
-        //    "# Bytes in all Heaps",
-        //    _processName,
-        //    true);
-
-        [JsonPropertyName("processCpuUsagePercent")]
-        public float ProcessCpuUsagePercent { get; set; } = s_processCpuCounter.NextValue();
-
-        [JsonPropertyName("processWorkingSet")]
-        public float ProcessWorkingSet { get; set; } = s_processWorkingSet.NextValue();
-
-        [JsonPropertyName("processWorkingSetPrivate")]
-        public float ProcessWorkingSetPrivate { get; set; } = s_processWorkingSetPrivate.NextValue();
-
-        [JsonPropertyName("processPrivateBytes")]
-        public float ProcessPrivateBytes { get; set; } = s_processPrivateBytes.NextValue();
-
-        [JsonPropertyName("processBytesInAllHeaps")]
-        public float? ProcessBytesInAllHeaps { get; set; } = s_processBytesInAllHeaps?.NextValue();
+        public static long TcpPortFilter;
+        public double ProcessCpu { get; set; } = s_currentProcess.TotalProcessorTime.TotalMilliseconds / Environment.ProcessorCount;
+        public long TotalAssignedMemoryBytes { get; set; } = s_currentProcess.WorkingSet64;
+        public long TotalGCBytes { get; set; } = GC.GetTotalMemory(false);
+        public long TcpConnections { get; set; } = UpdateTCPConnections();
+
+        public SystemHealthTelemetry(int port)
+        {
+            TcpPortFilter = port;
+        }
+
+
+        private static long UpdateTCPConnections()
+        {
+            IPGlobalProperties properties = IPGlobalProperties.GetIPGlobalProperties();
+            TcpConnectionInformation[] connections = properties.GetActiveTcpConnections();
+
+            long n = 0;
+            foreach (TcpConnectionInformation conn in connections)
+            {
+                if (TcpPortFilter != 0 && conn.RemoteEndPoint.Port != TcpPortFilter)
+                    continue;
+                if (conn.State == TcpState.Established)
+                    n++;
+            }
+            return n;
```

Comment:
Try Linq? Something like this:

```suggestion
            return connections
                .Where(c => TcpPortFilter != 0 && c.RemoteEndPoint.Port != TcpPortFilter)
                . Where(c => c.State == TcpState.Established)
                .Count();
```